In [1]:
import pandas as pd

In [46]:
data = pd.read_csv('title.basics.tsv',sep='\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [83]:
episodes[episodes.tconst.isin(['tt9759668'])]

tconst parentTconst seasonNumber episodeNumber
3874597  tt9759668    tt9647364            1            10

In [96]:
data[data.titleType.isin(['tvMiniSeries'])].tail(40)

tconst     titleType                              primaryTitle  \
5622883  tt9741242  tvMiniSeries                     Tokyo Nijusan Ku Onna   
5622939  tt9741366  tvMiniSeries            Stalked: Murder in Slow Motion   
5622966  tt9741426  tvMiniSeries                   How the Other Kids Live   
5623276  tt9742066  tvMiniSeries                                    Cycles   
5623303  tt9742120  tvMiniSeries                   Women of Adonai Project   
5623346  tt9742212  tvMiniSeries         Martin Clunes: Islands of America   
5623575  tt9742688  tvMiniSeries                           2 Girls, 1 Desk   
5624878  tt9745352  tvMiniSeries                                  Avec Moi   
5625873  tt9747402  tvMiniSeries                              The Enablers   
5625889  tt9747436  tvMiniSeries               My Most Embarrassing Moment   
5626209  tt9748112  tvMiniSeries                                     Handy   
5626239  tt9748172  tvMiniSeries                  The Ladies of Llangollen   
5626346  tt9748396  tvMiniSeries                           Fraud Squad NHS   
5626383  tt9748472  tvMiniSeries  Building the Channel Tunnel: 25 Years On   
5626553  tt9748820  tvMiniSeries                           Become Inspired   
5626673  tt9749064  tvMiniSeries                                 Law Women   
5626998  tt9749744  tvMiniSeries                           Secret Scotland   
5627192  tt9750148  tvMiniSeries                             Pistol Poetry   
5627300  tt9750380  tvMiniSeries                               Chess Queen   
5627921  tt9751650  tvMiniSeries                            Outlaw Platoon   
5627956  tt9751722  tvMiniSeries            Malditos cartunistas - a série   
5628476  tt9752814  tvMiniSeries                                Mom and Co   
5628549  tt9752980  tvMiniSeries                   Generation What? Europe   
5628573  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
5628581  tt9753046  tvMiniSeries                To Thine Own Self Be False   
5628589  tt9753064  tvMiniSeries  Nolan Jones - Les gardiens de la lumière   
5628668  tt9753222  tvMiniSeries                             Oh, So You Do   
5629643  tt9755302  tvMiniSeries                  Breaking the Fourth Wall   
5629980  tt9756050  tvMiniSeries                                    Psycho   
5630007  tt9756106  tvMiniSeries                         Nike op de Vlucht   
5630911  tt9757986  tvMiniSeries               Phil Spencer's Perfect Home   
5630957  tt9758080  tvMiniSeries                                    Olivia   
5630959  tt9758090  tvMiniSeries                                  Noturnas   
5631031  tt9758266  tvMiniSeries                         Histoires Courtes   
5631139  tt9758494  tvMiniSeries                              Brakkefolket   
5631147  tt9758514  tvMiniSeries                     Advent i prestegarden   
5631151  tt9758522  tvMiniSeries            Dave's 80,000 Point Chaos Army   
5631157  tt9758538  tvMiniSeries                                 Dalen vår   
5631165  tt9758556  tvMiniSeries              Berulfsens historiske perler   
5631439  tt9759178  tvMiniSeries                   The Real Tommy Robinson   

                                    originalTitle  isAdult startYear endYear  \
5622883                     Tokyo Nijusan Ku Onna        0      2019    2019   
5622939            Stalked: Murder in Slow Motion        0      2019    2019   
5622966                   How the Other Kids Live        0      2019    2019   
5623276                                    Cycles        0      2019    2019   
5623303                   Women of Adonai Project        0      2017      \N   
5623346         Martin Clunes: Islands of America        0      2019      \N   
5623575                           2 Girls, 1 Desk        0      2019      \N   
5624878                                  Avec Moi        0      2019      \N   
5625873                              The Enablers        0      2017      \N   
5625889      

In [57]:
data.titleType.unique()

array(['short', 'movie', 'tvMovie', 'tvSeries', 'tvEpisode', 'tvShort',
       'tvMiniSeries', 'tvSpecial', 'video', 'videoGame'], dtype=object)

In [53]:
data[data['titleType' == 'movie']].head(10)

SyntaxError: invalid syntax (<ipython-input-53-1ab9485413d8>, line 1)

In [3]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5631729 entries, 0 to 5631728
Data columns (total 9 columns):
tconst            object
titleType         object
primaryTitle      object
originalTitle     object
isAdult           int64
startYear         object
endYear           object
runtimeMinutes    object
genres            object
dtypes: int64(1), object(8)
memory usage: 386.7+ MB


what is the title basic tsv is about?
has 5631729 entries i.e 5.6 million entries of ?
relationship with genres is 1 : many


In [108]:
titles = data

In [109]:
titles.head()

tconst titleType            primaryTitle           originalTitle  \
0  tt0000001     short              Carmencita              Carmencita   
1  tt0000002     short  Le clown et ses chiens  Le clown et ses chiens   
2  tt0000003     short          Pauvre Pierrot          Pauvre Pierrot   
3  tt0000004     short             Un bon bock             Un bon bock   
4  tt0000005     short        Blacksmith Scene        Blacksmith Scene   

   isAdult startYear endYear runtimeMinutes                    genres  
0        0      1894      \N              1         Documentary,Short  
1        0      1892      \N              5           Animation,Short  
2        0      1892      \N              4  Animation,Comedy,Romance  
3        0      1892      \N             \N           Animation,Short  
4        0      1893      \N              1              Comedy,Short

In [19]:
titles['originalTitle'].nunique()

2931498

In [14]:
titles['titleType'].unique()

array(['short', 'movie', 'tvMovie', 'tvSeries', 'tvEpisode', 'tvShort',
       'tvMiniSeries', 'tvSpecial', 'video', 'videoGame'], dtype=object)

In [9]:
titles['tconst'].nunique()

5631729

In [17]:
titles['genres'].unique()
titles['genres'].head()

0           Documentary,Short
1             Animation,Short
2    Animation,Comedy,Romance
3             Animation,Short
4                Comedy,Short
Name: genres, dtype: object

In [110]:
titles['titleType'].value_counts()

tvEpisode       3874830
short            665570
movie            510690
video            222234
tvSeries         159635
tvMovie          125774
tvMiniSeries      24791
videoGame         22915
tvSpecial         16250
tvShort            9040
Name: titleType, dtype: int64

In [114]:
data1 = pd.read_csv('name.basics.tsv',sep='\t')

has 32 million entries

In [115]:
peopleWhoWorkInMovies = data1

In [119]:
peopleWhoWorkInMovies.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9206045 entries, 0 to 9206044
Data columns (total 6 columns):
nconst               object
primaryName          object
birthYear            object
deathYear            object
primaryProfession    object
knownForTitles       object
dtypes: object(6)
memory usage: 421.4+ MB


In [121]:
peopleWhoWorkInMovies.birthYear.value_counts()

\N      8730452
1980       8226
1979       8130
1978       8116
1981       8008
1982       7804
1977       7587
1976       7580
1983       7312
1975       7291
1974       7178
1984       7158
1970       7096
1971       6994
1972       6933
1973       6801
1985       6743
1986       6636
1969       6515
1968       6099
1967       5798
1965       5771
1987       5767
1964       5743
1966       5643
1963       5467
1988       5410
1962       5401
1961       5146
1947       5053
         ...   
0163          1
1413          1
1644          1
0770          1
1291          1
1733          1
1625          1
1597          1
1745          1
1677          1
1648          1
1631          1
1569          1
1765          1
1265          1
1468          1
1515          1
1562          1
1414          1
1532          1
1314          1
1586          1
1754          1
1207          1
1600          1
1752          1
0037          1
1415          1
0021          1
1681          1
Name: birthYear, Length:

In [118]:
data1 = pd.read_csv('title.principals.tsv',sep='\t')

In [122]:
principal = data1

In [123]:
principal.head()

tconst  ordering     nconst         category                      job  \
0  tt0000001         1  nm1588970             self                       \N   
1  tt0000001         2  nm0005690         director                       \N   
2  tt0000001         3  nm0374658  cinematographer  director of photography   
3  tt0000002         1  nm0721526         director                       \N   
4  tt0000002         2  nm1335271         composer                       \N   

    characters  
0  ["Herself"]  
1           \N  
2           \N  
3           \N  
4           \N

In [129]:
principal.tconst.value_counts(ascending=False).unique()

array([10,  9,  8,  7,  6,  5,  4,  3,  2,  1])

In [136]:
principal.job.value_counts(ascending=False)

\N                                                     26918368
producer                                                1876783
writer                                                   693415
creator                                                  496450
written by                                               336796
story                                                    159174
director of photography                                  143642
screenplay                                               136989
head writer                                               95814
adaptation                                                83734
co-director                                               71186
novel                                                     42195
teleplay                                                  41150
series director                                           37010
dialogue                                                  36625
script                                  

In [130]:
principal.categorytegory.value_counts(ascending=False)

actor                  7668975
actress                5560782
self                   5495951
writer                 4100995
director               3611035
producer               1918242
composer               1187555
cinematographer        1154846
editor                 1078652
production_designer     261679
archive_footage         187035
archive_sound             1949
Name: category, dtype: int64

In [26]:
principal['category'].unique()

array(['self', 'director', 'cinematographer', 'composer', 'producer',
       'editor', 'actor', 'actress', 'writer', 'production_designer',
       'archive_footage', 'archive_sound'], dtype=object)

In [29]:
principal['characters'].unique()

array(['["Herself"]', '\\N', '["Blacksmith"]', ..., '["Andy Westfall"]',
       '["Yirga"]', '["Mike van Loo"]'], dtype=object)

shows for a movie the nconst has played what character in the movie

In [30]:
principal['tconst'].nunique()

5047848

In [37]:
data1 = pd.read_csv('title.crew.tsv',sep='\t')

In [38]:
crew = data1

In [39]:
crew.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5723534 entries, 0 to 5723533
Data columns (total 3 columns):
tconst       object
directors    object
writers      object
dtypes: object(3)
memory usage: 131.0+ MB


In [100]:
crew.head()

tconst  directors writers
0  tt0000001  nm0005690      \N
1  tt0000002  nm0721526      \N
2  tt0000003  nm0721526      \N
3  tt0000004  nm0721526      \N
4  tt0000005  nm0005690      \N

In [41]:
data1 = pd.read_csv('title.ratings.tsv',sep='\t')

In [42]:
ratings = data1

In [43]:
ratings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 917181 entries, 0 to 917180
Data columns (total 3 columns):
tconst           917181 non-null object
averageRating    917181 non-null float64
numVotes         917181 non-null int64
dtypes: float64(1), int64(1), object(1)
memory usage: 21.0+ MB


In [44]:
ratings.head()

tconst  averageRating  numVotes
0  tt0000001            5.8      1478
1  tt0000002            6.4       176
2  tt0000003            6.6      1108
3  tt0000004            6.4       107
4  tt0000005            6.2      1813

In [45]:
ratings.describe()

averageRating      numVotes
count  917181.000000  9.171810e+05
mean        6.918489  9.705193e+02
std         1.401053  1.553299e+04
min         1.000000  5.000000e+00
25%         6.200000  9.000000e+00
50%         7.100000  2.000000e+01
75%         7.900000  7.700000e+01
max        10.000000  2.064369e+06

In [93]:

data1 = pd.read_csv('title.akas.tsv',sep='\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (7) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [94]:
alsoKnownAs = data1

In [132]:
alsoKnownAs.titleId.value_counts(ascending=False)

tt8185052    129
tt0168366    115
tt8052676     88
tt0076759     79
tt8232202     78
tt8187756     77
tt0060196     73
tt0057012     68
tt0080684     67
tt0029583     64
tt0088247     63
tt0086190     63
tt0071562     62
tt2488496     61
tt0058461     61
tt0064116     60
tt0068646     59
tt0080339     59
tt0082971     59
tt0044706     58
tt0111161     58
tt0052357     57
tt0245429     57
tt0047478     57
tt0096874     57
tt0367882     57
tt0028950     57
tt0064208     56
tt0047396     56
tt0073195     56
            ... 
tt1604408      1
tt6025896      1
tt0745627      1
tt4866710      1
tt6760312      1
tt3788922      1
tt0044678      1
tt0907126      1
tt3632686      1
tt0590394      1
tt5424634      1
tt2110365      1
tt3825272      1
tt7283236      1
tt0286242      1
tt7183820      1
tt6750864      1
tt0538874      1
tt0995145      1
tt1016822      1
tt5901648      1
tt2077325      1
tt5054554      1
tt7187408      1
tt7148720      1
tt1864082      1
tt0977207      1
tt2083219     

In [47]:
alsoKnownAs.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3484661 entries, 0 to 3484660
Data columns (total 8 columns):
titleId            object
ordering           int64
title              object
region             object
language           object
types              object
attributes         object
isOriginalTitle    object
dtypes: int64(1), object(7)
memory usage: 212.7+ MB


In [48]:
alsoKnownAs.head()

titleId  ordering                      title region language  \
0  tt0000001         1  Carmencita - spanyol tánc     HU       \N   
1  tt0000001         2                 Καρμενσίτα     GR       \N   
2  tt0000001         3                 Карменсита     RU       \N   
3  tt0000001         4                 Carmencita     US       \N   
4  tt0000001         5                 Carmencita     \N       \N   

         types attributes isOriginalTitle  
0  imdbDisplay         \N               0  
1           \N         \N               0  
2           \N         \N               0  
3           \N         \N               0  
4     original         \N               1

In [49]:
alsoKnownAs['titleId'].nunique()

2023643

In [133]:
alsoKnownAs['region'].value_counts()

US      906267
\N      404744
GB      253700
FR      139897
JP      115591
ES      105772
DE      103437
XWW     101703
CA       87597
AR       81500
IT       78541
AU       73243
BR       73243
GR       63540
FI       56416
PT       51747
HU       50524
RU       45531
IN       45248
PL       43039
SE       42271
XWG      41967
MX       41726
DK       40529
NL       27873
TR       27175
SUHH     23834
BE       22870
AT       19456
BG       18861
         ...  
AQ           5
TV           5
CK           4
AS           4
SB           4
XPI          4
GF           4
EH           3
VA           3
WF           3
GD           3
SC           3
GQ           3
XNA          2
ZRCD         2
JE           2
WS           2
SH           2
MS           2
ST           2
KN           2
KI           2
NU           1
XAU          1
AI           1
PW           1
VG           1
MP           1
DJ           1
NR           1
Name: region, Length: 245, dtype: int64

In [134]:
alsoKnownAs['types'].value_counts()

\N                  2550556
imdbDisplay          427658
original             356622
alternative           52300
working               39566
dvd                   18617
tv                    14578
festival              12593
video                 12149
tvvideo                  6
videoworking             5
tvworking                4
dvdimdbDisplay           3
festivalworking          2
dvdworking               1
dvdvideo                 1
Name: types, dtype: int64

In [135]:
alsoKnownAs['language'].value_counts()

\N     3230043
en      138519
ru       21872
fr       18854
bg       17933
tr       15479
sv        6632
he        5430
qbn       4649
sr        3430
cmn       2952
yue       2653
ca        1838
fa        1786
hi        1424
hr        1335
es        1118
tl         885
cs         795
de         785
sl         723
ar         537
nl         536
qbp        496
ta         496
ml         475
te         329
sk         228
bn         224
ur         223
        ...   
da          10
no           7
gd           6
ps           5
uz           5
xh           5
mi           4
vi           4
sd           3
st           3
kk           3
be           2
is           2
ko           2
wo           2
zh           2
tg           2
tn           2
ku           1
goh          1
rm           1
prs          1
rn           1
sq           1
fro          1
myv          1
nqo          1
iu           1
cr           1
ky           1
Name: language, Length: 95, dtype: int64

In [73]:
data1 = pd.read_csv('title.episode.tsv',sep='\t')

In [75]:
episodes = data1

In [59]:
episodes.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3874634 entries, 0 to 3874633
Data columns (total 4 columns):
tconst           object
parentTconst     object
seasonNumber     object
episodeNumber    object
dtypes: object(4)
memory usage: 118.2+ MB


In [78]:
episodes[episodes.tconst.isin(['tt9759726'])]

tconst parentTconst seasonNumber episodeNumber
3874610  tt9759726    tt8041852            1            12

In [60]:
episodes.head()

tconst parentTconst seasonNumber episodeNumber
0  tt0041951    tt0041038            1             9
1  tt0042816    tt0989125            1            17
2  tt0042889    tt0989125           \N            \N
3  tt0043426    tt0040051            3            42
4  tt0043631    tt0989125            2            16

In [71]:
episodes.groupby(['parentTconst'])['tconst'].count()

parentTconst
tt0038276       1
tt0039122       5
tt0039123     587
tt0039125      12
tt0040021      65
tt0040027       2
tt0040028      14
tt0040030       1
tt0040032       1
tt0040033       4
tt0040034       4
tt0040036      82
tt0040038       1
tt0040039      10
tt0040040       1
tt0040041     261
tt0040042      78
tt0040045       4
tt0040048    1003
tt0040049     266
tt0040051     467
tt0040052       2
tt0040053    1070
tt0040055       1
tt0040057      21
tt0040058       1
tt0040993      18
tt0040994      23
tt0040995       1
tt0040997       1
             ... 
tt9741848       1
tt9742390      53
tt9742810      26
tt9742936       1
tt9743578       2
tt9743630       4
tt9744774       1
tt9745284       5
tt9747398      21
tt9747414       2
tt9748016       4
tt9748396       5
tt9748472       2
tt9748820       3
tt9749192      19
tt9749378      13
tt9749744       3
tt9749918       1
tt9749982       3
tt9750148       5
tt9751532       1
tt9753030      12
tt9753256       1
tt9755000      

In [78]:
episodes[episodes.parentTconst == 'tt9758034']

tconst parentTconst seasonNumber episodeNumber
3874066  tt9758294    tt9758034            1             1
3874072  tt9758306    tt9758034            1             2
3874081  tt9758326    tt9758034            1             3
3874086  tt9758336    tt9758034            1             4
3874097  tt9758360    tt9758034            1             5
3874102  tt9758370    tt9758034            1             6
3874108  tt9758384    tt9758034            1             7
3874111  tt9758392    tt9758034            1             8
3874116  tt9758402    tt9758034            1             9
3874120  tt9758412    tt9758034            1            10
3874122  tt9758416    tt9758034            1            11
3874123  tt9758418    tt9758034            1            12
3874125  tt9758422    tt9758034            1            13
3874128  tt9758428    tt9758034            1            14
3874129  tt9758430    tt9758034            1            15
3874131  tt9758434    tt9758034            1            16
3874133  tt9758438    tt9758034            1            17
3874135  tt9758442    tt9758034            1            18

In [88]:
ratings[ratings.tconst.isin(episodes.parentTconst.unique())].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 60983 entries, 19682 to 916396
Data columns (total 3 columns):
tconst           60983 non-null object
averageRating    60983 non-null float64
numVotes         60983 non-null int64
dtypes: float64(1), int64(1), object(1)
memory usage: 1.9+ MB


In [92]:
episodes.parentTconst.value_counts(ascending=False)

tt0058796    10541
tt0053494     9667
tt1985601     9502
tt0088580     8911
tt0068120     8697
tt0283794     8651
tt0283767     8104
tt0159881     7859
tt0092325     7636
tt0088512     7633
tt0094481     7081
tt0072584     6795
tt0055708     6746
tt0103434     6695
tt0068069     6420
tt0069658     6353
tt0108977     6043
tt0096597     5903
tt0056758     5870
tt0138228     5584
tt0371487     5186
tt0108979     4664
tt0103569     4656
tt1152300     4601
tt6769314     4601
tt0057741     4443
tt0106053     4238
tt0170980     4069
tt0044298     4052
tt2070564     3978
             ...  
tt2731560        1
tt6113146        1
tt5766950        1
tt7654048        1
tt3668324        1
tt6539592        1
tt4636670        1
tt6233072        1
tt7166874        1
tt6815958        1
tt9071112        1
tt5872906        1
tt2492910        1
tt7707776        1
tt3771468        1
tt6257772        1
tt2300026        1
tt5092982        1
tt4519414        1
tt0268817        1
tt4549714        1
tt0279574   

In [2]:
import pandas as pd
person = pd.read_csv('name.basics.tsv',sep='\t')

In [3]:
person.head()

nconst      primaryName birthYear deathYear  \
0  nm0000001     Fred Astaire      1899      1987   
1  nm0000002    Lauren Bacall      1924      2014   
2  nm0000003  Brigitte Bardot      1934        \N   
3  nm0000004     John Belushi      1949      1982   
4  nm0000005   Ingmar Bergman      1918      2007   

                primaryProfession                           knownForTitles  
0  soundtrack,actor,miscellaneous  tt0072308,tt0053137,tt0043044,tt0050419  
1              actress,soundtrack  tt0038355,tt0071877,tt0037382,tt0117057  
2     actress,soundtrack,producer  tt0059956,tt0057345,tt0054452,tt0049189  
3         actor,writer,soundtrack  tt0078723,tt0072562,tt0080455,tt0077975  
4           writer,director,actor  tt0069467,tt0050976,tt0050986,tt0083922

In [28]:
personKnownForTitles.head(0)

nconst                           knownForTitles
0   nm0000001  tt0072308,tt0053137,tt0043044,tt0050419
1   nm0000002  tt0038355,tt0071877,tt0037382,tt0117057
2   nm0000003  tt0059956,tt0057345,tt0054452,tt0049189
3   nm0000004  tt0078723,tt0072562,tt0080455,tt0077975
4   nm0000005  tt0069467,tt0050976,tt0050986,tt0083922
5   nm0000006  tt0036855,tt0038787,tt0038109,tt0071877
6   nm0000007  tt0037382,tt0034583,tt0043265,tt0033870
7   nm0000008  tt0047296,tt0078788,tt0068646,tt0070849
8   nm0000009  tt0061184,tt0059749,tt0087803,tt0057877
9   nm0000010  tt0029870,tt0055256,tt0035575,tt0031867
10  nm0000011  tt0034167,tt0044706,tt0035211,tt0027996
11  nm0000012  tt0031210,tt0035140,tt0056687,tt0042192
12  nm0000013  tt0060463,tt0055100,tt0053172,tt0048317
13  nm0000014  tt0041452,tt0031381,tt0029843,tt0040806
14  nm0000015  tt0048028,tt0045458,tt0048545,tt0049261
15  nm0000016  tt0069946,tt0091763,tt0080610,tt0096320
16  nm0000017  tt0052311,tt0040367,tt0055031,tt0051201
17  nm0000018  tt0049456,tt0052365,tt0054331,tt0080736
18  nm0000019  tt0056801,tt0047528,tt0053779,tt0071129
19  nm0000020  tt0032551,tt0050083,tt0064116,tt0082846
20  nm0000021  tt0039504,tt0036969,tt0032976,tt0034248
21  nm0000022  tt0025316,tt0031381,tt0026752,tt0052278
22  nm0000023  tt0037059,tt0047522,tt0055031,tt0032138
23  nm0000024  tt0117631,tt0045943,tt0071877,tt0082031
24  nm0000025  tt0117731,tt0100802,tt0119488,tt0112715
25  nm0000026  tt0053125,tt0056923,tt0036613,tt0048728
26  nm0000027  tt0051739,tt0041546,tt0050212,tt0076759
27  nm0000028  tt0038559,tt0040552,tt0040525,tt0036723
28  nm0000029  tt0077800,tt0074802,tt0110138,tt0102122
29  nm0000030  tt0058385,tt0056923,tt0054698,tt0046250
30  nm0000031  tt0063227,tt0082846,tt0032904,tt0043265
31  nm0000032  tt0063442,tt0070723,tt0049833,tt0052618
32  nm0000033  tt0030341,tt0054215,tt0052357,tt0053125
33  nm0000034  tt0043014,tt0046359,tt0074958,tt0065214
34  nm0000035  tt0120647,tt0499549,tt0120746,tt0120338
35  nm0000036  tt0016332,tt0015324,tt0015863,tt0017925
36  nm0000037  tt0045152,tt0043278,tt0041944,tt0041716
37  nm0000038  tt0046912,tt0048728,tt0044706,tt0047396
38  nm0000039  tt0049408,tt0050105,tt0045793,tt0039192
39  nm0000040  tt0072684,tt0120663,tt0062622,tt0066921
40  nm0000041  tt0080979,tt0047478,tt0089881,tt0051808
41  nm0000042  tt0038369,tt0035432,tt0046303,tt0043292
42  nm0000043  tt0034798,tt0034881,tt0038369,tt0034240
43  nm0000044  tt0050468,tt0045793,tt0059825,tt0053793
44  nm0000045  tt0068767,tt0068935,tt0070034,tt0077594
45  nm0000046  tt0031381,tt0055382,tt0044081,tt0033238
46  nm0000047  tt0076085,tt0054749,tt0060121,tt0058335
47  nm0000048  tt0033870,tt0036613,tt0034583,tt0022100
48  nm0000049  tt0057413,tt0383216,tt0085101,tt0054698
49  nm0000050  tt0028772,tt0023969,tt0026778,tt0022158

In [77]:
personKnownForTitles = person[['nconst','knownForTitles']]
frame = pd.DataFrame(personKnownForTitles.knownForTitles.str.split(',').tolist(), index=personKnownForTitles.nconst).stack()
frame = frame.reset_index()[['nconst',0]]
frame.columns = ['nconst','knownForTitles']

In [78]:
frame.head(10)

nconst knownForTitles
0  nm0000001      tt0072308
1  nm0000001      tt0053137
2  nm0000001      tt0043044
3  nm0000001      tt0050419
4  nm0000002      tt0038355
5  nm0000002      tt0071877
6  nm0000002      tt0037382
7  nm0000002      tt0117057
8  nm0000003      tt0059956
9  nm0000003      tt0057345

In [79]:
frame.to_csv('knownForTitles.tsv',sep='\t')

In [41]:
personKnownForTitles = person[['nconst','knownForTitles']]

In [ ]:
frame = pd.concat([pd.Series(row['nconst'], row['knownForTitles'].split(','))              
                    for _, row in personKnownForTitles.iterrows()])

In [42]:
frame = pd.concat([pd.Series(row['nconst'], row['knownForTitles'].split(','))              
                    for _, row in personKnownForTitles.iterrows()])

KeyboardInterrupt: 

In [36]:
frame.head()

tt0072308    nm0000001
tt0053137    nm0000001
tt0043044    nm0000001
tt0050419    nm0000001
tt0038355    nm0000002
dtype: object

In [17]:
pd.concat([pd.Series(row['var2'], row['var1'].split(','))              
                    for _, row in a.iterrows()]).reset_index()

index  0
0     a  1
1     b  1
2     c  1
3     d  2
4     e  2
5     f  2

In [85]:
data = pd.read_csv('title.basics.tsv',sep='\t')
temp = data[['tconst','genres']]
temp.dropna(inplace=True)
frame = pd.DataFrame(temp.genres.str.split(',').tolist(), index=temp.tconst).stack()
frame = frame.reset_index()[['tconst',0]]
frame.columns = ['tconst','genres']
frame.to_csv('title_genres.tsv', sep='\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


TypeError: 'Series' objects are mutable, thus they cannot be hashed

In [81]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5631729 entries, 0 to 5631728
Data columns (total 9 columns):
tconst            object
titleType         object
primaryTitle      object
originalTitle     object
isAdult           int64
startYear         object
endYear           object
runtimeMinutes    object
genres            object
dtypes: int64(1), object(8)
memory usage: 386.7+ MB


In [83]:
frame.head()

tconst titleType  level_2            0
0  tt0000001     short        0  Documentary
1  tt0000001     short        1        Short
2  tt0000002     short        0    Animation
3  tt0000002     short        1        Short
4  tt0000003     short        0    Animation

In [126]:
data = pd.read_csv('name.basics.tsv',sep='\t')
temp = data[['nconst','primaryProfession']]
temp.dropna(inplace=True)
frame = pd.DataFrame(temp.primaryProfession.str.split(',').tolist(), index=temp.nconst).stack()
frame = frame.reset_index()[['nconst',0]]
frame.columns = ['nconst','primaryProfession']
frame.to_csv('namesProfessions.tsv', sep='\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  This is separate from the ipykernel package so we can avoid doing imports until


In [130]:
primaryProfession = pd.read_csv('namesProfessions.tsv',sep='\t')

In [134]:
primaryProfession.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10319370 entries, 0 to 10319369
Data columns (total 3 columns):
Unnamed: 0           int64
nconst               object
primaryProfession    object
dtypes: int64(1), object(2)
memory usage: 236.2+ MB


In [140]:
file = open('testfile.sql','w') 
file.close()
file = open('testfile.sql','a') 
for index, row in primaryProfession.iterrows():
    file.writelines('Insert into table_ values(\''+ row['nconst'] + '\',\'' + row['primaryProfession']+ '\');\n')
     
file.close()    

In [2]:
data = pd.read_csv('name.basics.tsv',sep='\t')

In [3]:
data.primaryName.nunique()

7244675

In [6]:
data[['primaryName','birthYear']].nunique()

primaryName    7244675
birthYear          473
dtype: int64

In [23]:
names = data[['primaryName','birthYear','deathYear']].head(10)

In [24]:
names.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10 entries, 0 to 9
Data columns (total 3 columns):
primaryName    10 non-null object
birthYear      10 non-null object
deathYear      10 non-null object
dtypes: object(3)
memory usage: 320.0+ bytes


In [28]:
names.head(10)

primaryName birthYear deathYear
0     Fred Astaire      1899      1987
1    Lauren Bacall      1924      2014
2  Brigitte Bardot      1934        \N
3     John Belushi      1949      1982
4   Ingmar Bergman      1918      2007
5   Ingrid Bergman      1915      1982
6  Humphrey Bogart      1899      1957
7    Marlon Brando      1924      2004
8   Richard Burton      1925      1984
9     James Cagney      1899      1986

In [42]:
d = {'col1': [1, 2,1,1,1], 'col2': [3, 4,2,1,1]}
df = pd.DataFrame(data=d)

In [ ]:
epid

In [43]:
df.head()

col1  col2
0     1     3
1     2     4
2     1     2
3     1     1
4     1     1

In [44]:
df.drop_duplicates(subset=['col1','col2'],inplace=True)

In [45]:
df.head()

col1  col2
0     1     3
1     2     4
2     1     2
3     1     1

In [27]:
names.drop_duplicates(inplace = True)

In [21]:
names.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7354516 entries, 0 to 9206043
Data columns (total 3 columns):
primaryName    object
birthYear      object
deathYear      object
dtypes: object(3)
memory usage: 224.4+ MB


In [22]:
9206045 - 7354516

1851529

In [ ]:
names = pd.DataFrame({''})

In [15]:
names.drop_duplicates(inplace=True)

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/ipykernel_launcher.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: http://pandas.pydata.org/pandas-docs/stable/indexing.html#indexing-view-versus-copy
  """Entry point for launching an IPython kernel.


In [17]:
names.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 7354516 entries, 0 to 9206043
Data columns (total 3 columns):
primaryName    object
birthYear      object
deathYear      object
dtypes: object(3)
memory usage: 224.4+ MB


In [125]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9206045 entries, 0 to 9206044
Data columns (total 6 columns):
nconst               object
primaryName          object
birthYear            object
deathYear            object
primaryProfession    object
knownForTitles       object
dtypes: object(6)
memory usage: 421.4+ MB


In [1]:
import pandas as pd

In [2]:
titles = pd.read_csv('title.basics.tsv',sep='\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [3]:
titles.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5631729 entries, 0 to 5631728
Data columns (total 9 columns):
tconst            object
titleType         object
primaryTitle      object
originalTitle     object
isAdult           int64
startYear         object
endYear           object
runtimeMinutes    object
genres            object
dtypes: int64(1), object(8)
memory usage: 386.7+ MB


In [116]:
titles.runtimeMinutes.nunique()

808

In [127]:
short = titles

In [176]:
short = titles[titles.titleType.isin(['video'])]

In [118]:
short[short.isAdult == 1].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 8546 entries, 59188 to 5628522
Data columns (total 9 columns):
tconst            8546 non-null object
titleType         8546 non-null object
primaryTitle      8546 non-null object
originalTitle     8546 non-null object
isAdult           8546 non-null int64
startYear         8546 non-null object
endYear           8546 non-null object
runtimeMinutes    8546 non-null object
genres            8546 non-null object
dtypes: int64(1), object(8)
memory usage: 667.7+ KB


In [119]:
short.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 510690 entries, 8 to 5631715
Data columns (total 9 columns):
tconst            510690 non-null object
titleType         510690 non-null object
primaryTitle      510690 non-null object
originalTitle     510690 non-null object
isAdult           510690 non-null int64
startYear         510690 non-null object
endYear           510690 non-null object
runtimeMinutes    510690 non-null object
genres            510690 non-null object
dtypes: int64(1), object(8)
memory usage: 39.0+ MB


In [177]:
short.tail()

tconst titleType                          primaryTitle  \
5631294  tt9758870     video         MasterCard #AcceptanceMatters   
5631460  tt9759224     video             Signing Time! Sing & Sign   
5631681  tt9759688     video   Still Shaving After All These Beers   
5631687  tt9759704     video                               Mistero   
5631710  tt9759790     video  Kamikaze Premium 005: Nanami Komachi   

                                originalTitle  isAdult startYear endYear  \
5631294         MasterCard #AcceptanceMatters        0      2018      \N   
5631460             Signing Time! Sing & Sign        0      2005      \N   
5631681   Still Shaving After All These Beers        0      2017      \N   
5631687                               Mistero        0      2018      \N   
5631710  Kamikaze Premium 005: Nanami Komachi        1      2006      \N   

        runtimeMinutes       genres  
5631294             \N        Short  
5631460             \N       Family  
5631681              4  Music,Short  
5631687             70  Documentary  
5631710             91        Adult

In [74]:
short[short.endYear.isin(['\\N'])].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 22915 entries, 82589 to 5629832
Data columns (total 9 columns):
tconst            22915 non-null object
titleType         22915 non-null object
primaryTitle      22915 non-null object
originalTitle     22915 non-null object
isAdult           22915 non-null int64
startYear         22915 non-null object
endYear           22915 non-null object
runtimeMinutes    22915 non-null object
genres            22915 non-null object
dtypes: int64(1), object(8)
memory usage: 1.7+ MB


In [121]:
short[short.primaryTitle != short.originalTitle].tail()

tconst titleType  \
5620222  tt9735672     movie   
5622533  tt9740508     movie   
5625440  tt9746500     movie   
5626108  tt9747894     movie   
5626911  tt9749570     movie   

                                              primaryTitle  \
5620222  Doraemon: Nobita's Chronicle of the Moon Explo...   
5622533                                        Wedding Day   
5625440                                              Earth   
5626108                                         Voice Over   
5626911                          Heimat Is a Space in Time   

                                     originalTitle  isAdult startYear endYear  \
5620222  Eiga Doraemon: Nobita no Getsumen Tansaki        0      2019      \N   
5622533                                 Totsugu hi        0      1956      \N   
5625440                                       Erde        0      2019      \N   
5626108                   Le'at Yoter - Voice Over        0      2018      \N   
5626911               Heimat ist ein Raum aus Zeit        0      2019      \N   

        runtimeMinutes                      genres  
5620222             \N  Adventure,Animation,Comedy  
5622533             89                       Drama  
5625440            115                 Documentary  
5626108             85                       Drama  
5626911            218                 Documentary

In [135]:
m[m.titleId.isin(short[short.primaryTitle != short.originalTitle])]['tconst'].tail(100)

Series([], Name: tconst, dtype: object)

In [129]:
short[short.tconst== 'tt0000098']

tconst titleType                   primaryTitle  \
96  tt0000098     short  Place de l'Opéra, Second View   

                  originalTitle  isAdult startYear endYear runtimeMinutes  \
96  Place de l'Opéra, 2e aspect        0      1896      \N             \N   

               genres  
96  Documentary,Short

In [143]:
cast[cast.tconst=='tt9451854']

tconst  ordering     nconst         category job  \
31464887  tt9451854         1  nm1169062            actor  \N   
31464888  tt9451854         2  nm0639011            actor  \N   
31464889  tt9451854         3  nm1516233            actor  \N   
31464890  tt9451854         4  nm6179754             self  \N   
31464891  tt9451854         5  nm1515269         director  \N   
31464892  tt9451854         6  nm1841245  cinematographer  \N   
31464893  tt9451854         7  nm2569852           editor  \N   

                              characters  
31464887                        ["Riku"]  
31464888                       ["Ransu"]  
31464889                         ["Eno"]  
31464890  ["Himself - kaupunginjohtaja"]  
31464891                              \N  
31464892                              \N  
31464893                              \N

In [172]:
series[series.parentTconst.isin(e.tconst)].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 89969 entries, 52 to 3874627
Data columns (total 4 columns):
tconst           89969 non-null object
parentTconst     89969 non-null object
seasonNumber     89969 non-null object
episodeNumber    89969 non-null object
dtypes: object(4)
memory usage: 3.4+ MB


In [168]:
e = titles[titles.titleType == 'tvMiniSeries']

In [170]:
m[m.titleId.isin(e.tconst)].tail()

TypeError: '<' not supported between instances of 'float' and 'str'

In [171]:
e.tail(100)

tconst     titleType  \
5607014  tt9708262  tvMiniSeries   
5607140  tt9708550  tvMiniSeries   
5608050  tt9710382  tvMiniSeries   
5608210  tt9710744  tvMiniSeries   
5608563  tt9711460  tvMiniSeries   
5609329  tt9713016  tvMiniSeries   
5609355  tt9713072  tvMiniSeries   
5609486  tt9713334  tvMiniSeries   
5609815  tt9714002  tvMiniSeries   
5609854  tt9714080  tvMiniSeries   
5610132  tt9714658  tvMiniSeries   
5610605  tt9715654  tvMiniSeries   
5610799  tt9716070  tvMiniSeries   
5610919  tt9716320  tvMiniSeries   
5611216  tt9716950  tvMiniSeries   
5612025  tt9718602  tvMiniSeries   
5612028  tt9718610  tvMiniSeries   
5612502  tt9719586  tvMiniSeries   
5613714  tt9722052  tvMiniSeries   
5613767  tt9722164  tvMiniSeries   
5613851  tt9722338  tvMiniSeries   
5613862  tt9722362  tvMiniSeries   
5614083  tt9722836  tvMiniSeries   
5614302  tt9723326  tvMiniSeries   
5614444  tt9723636  tvMiniSeries   
5614478  tt9723710  tvMiniSeries   
5614730  tt9724268  tvMiniSeries   
5614756  tt9724320  tvMiniSeries   
5614796  tt9724402  tvMiniSeries   
5614953  tt9724740  tvMiniSeries   
...            ...           ...   
5626209  tt9748112  tvMiniSeries   
5626239  tt9748172  tvMiniSeries   
5626346  tt9748396  tvMiniSeries   
5626383  tt9748472  tvMiniSeries   
5626553  tt9748820  tvMiniSeries   
5626673  tt9749064  tvMiniSeries   
5626998  tt9749744  tvMiniSeries   
5627192  tt9750148  tvMiniSeries   
5627300  tt9750380  tvMiniSeries   
5627921  tt9751650  tvMiniSeries   
5627956  tt9751722  tvMiniSeries   
5628476  tt9752814  tvMiniSeries   
5628549  tt9752980  tvMiniSeries   
5628573  tt9753030  tvMiniSeries   
5628581  tt9753046  tvMiniSeries   
5628589  tt9753064  tvMiniSeries   
5628668  tt9753222  tvMiniSeries   
5629643  tt9755302  tvMiniSeries   
5629980  tt9756050  tvMiniSeries   
5630007  tt9756106  tvMiniSeries   
5630911  tt9757986  tvMiniSeries   
5630957  tt9758080  tvMiniSeries   
5630959  tt9758090  tvMiniSeries   
5631031  tt9758266  tvMiniSeries   
5631139  tt9758494  tvMiniSeries   
5631147  tt9758514  tvMiniSeries   
5631151  tt9758522  tvMiniSeries   
5631157  tt9758538  tvMiniSeries   
5631165  tt9758556  tvMiniSeries   
5631439  tt9759178  tvMiniSeries   

                                              primaryTitle  \
5607014                            A History of Loneliness   
5607140                                      Country Music   
5608050                                            Baliboo   
5608210                                The Senapatis Vol-1   
5608563                                              Punga   
5609329                           Inside Brookelyn's World   
5609355                               O Baldio Som de Deus   
5609486                                    Me ton Loukiano   
5609815                           The Penultimate BronyCon   
5609854                                      Wantrepreneur   
5610132                                   Wind in the Face   
5610605                                      The Red Cloth   
5610799                       WWI: The War to End all Wars   
5610919                                                WTF   
5611216                       Monty Don's Japanese Gardens   
5612025                            Autopsy: Life and Death   
5612028                                 Fora de experiente   
5612502                                          La Vallée   
5613714                         Wild Times (Animal Planet)   
5613767                                             Shoed!   
5613851                                    Newclear Family   
5613862                                         Lou Howard   
5614083                                    All the Bitches   
5614302                                        N.I.N.J.A.S   
5614444                                 Göthenburgo Begins   
5614478                             Suki Desu Suzuki-kun!!   
5614730                                      The Professor   
5614756                                   C

In [76]:
titles = pd.read_csv('title.basics.tsv',sep='\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [91]:
titles.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5631729 entries, 0 to 5631728
Data columns (total 9 columns):
tconst            object
titleType         object
primaryTitle      object
originalTitle     object
isAdult           int64
startYear         object
endYear           object
runtimeMinutes    object
genres            object
dtypes: int64(1), object(8)
memory usage: 386.7+ MB


In [77]:
regional = pd.read_csv('title.akas.tsv',sep = '\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (7) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [78]:
regional.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3484661 entries, 0 to 3484660
Data columns (total 8 columns):
titleId            object
ordering           int64
title              object
region             object
language           object
types              object
attributes         object
isOriginalTitle    object
dtypes: int64(1), object(7)
memory usage: 212.7+ MB


In [132]:
m = pd.merge(titles,regional,how='left',left_on='tconst' , right_on='titleId')

In [173]:
sr = pd.merge(e,series,how='inner',left_on='tconst' , right_on='parentTconst')

In [175]:
sr.tail(100)

tconst_x     titleType                              primaryTitle  \
89869  tt9733350  tvMiniSeries                             Lets Find Out   
89870  tt9733350  tvMiniSeries                             Lets Find Out   
89871  tt9733350  tvMiniSeries                             Lets Find Out   
89872  tt9733350  tvMiniSeries                             Lets Find Out   
89873  tt9733350  tvMiniSeries                             Lets Find Out   
89874  tt9733350  tvMiniSeries                             Lets Find Out   
89875  tt9733350  tvMiniSeries                             Lets Find Out   
89876  tt9733350  tvMiniSeries                             Lets Find Out   
89877  tt9733350  tvMiniSeries                             Lets Find Out   
89878  tt9733350  tvMiniSeries                             Lets Find Out   
89879  tt9733350  tvMiniSeries                             Lets Find Out   
89880  tt9733350  tvMiniSeries                             Lets Find Out   
89881  tt9733350  tvMiniSeries                             Lets Find Out   
89882  tt9733350  tvMiniSeries                             Lets Find Out   
89883  tt9733350  tvMiniSeries                             Lets Find Out   
89884  tt9733350  tvMiniSeries                             Lets Find Out   
89885  tt9733350  tvMiniSeries                             Lets Find Out   
89886  tt9733350  tvMiniSeries                             Lets Find Out   
89887  tt9733350  tvMiniSeries                             Lets Find Out   
89888  tt9733350  tvMiniSeries                             Lets Find Out   
89889  tt9734446  tvMiniSeries  Scener ur ett inte alltför bra äktenskap   
89890  tt9734446  tvMiniSeries  Scener ur ett inte alltför bra äktenskap   
89891  tt9734446  tvMiniSeries  Scener ur ett inte alltför bra äktenskap   
89892  tt9734446  tvMiniSeries  Scener ur ett inte alltför bra äktenskap   
89893  tt9734446  tvMiniSeries  Scener ur ett inte alltför bra äktenskap   
89894  tt9734446  tvMiniSeries  Scener ur ett inte alltför bra äktenskap   
89895  tt9734446  tvMiniSeries  Scener ur ett inte alltför bra äktenskap   
89896  tt9734446  tvMiniSeries  Scener ur ett inte alltför bra äktenskap   
89897  tt9734446  tvMiniSeries  Scener ur ett inte alltför bra äktenskap   
89898  tt9734446  tvMiniSeries  Scener ur ett inte alltför bra äktenskap   
...          ...           ...                                       ...   
89939  tt9750148  tvMiniSeries                             Pistol Poetry   
89940  tt9750148  tvMiniSeries                             Pistol Poetry   
89941  tt9750148  tvMiniSeries                             Pistol Poetry   
89942  tt9750148  tvMiniSeries                             Pistol Poetry   
89943  tt9750148  tvMiniSeries                             Pistol Poetry   
89944  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89945  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89946  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89947  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89948  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89949  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89950  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89951  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89952  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89953  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89954  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89955  tt9753030  tvMiniSeries                     Untitled J&GS Spinoff   
89956  tt9756050  tvMiniSeries                                    Psycho   
89957  tt9756050  tvMiniSeries                                    Psycho   
89958  tt9756050  tvMiniSeries                                    Psycho   
89959  tt9756050  tvMiniSeries                          

In [174]:
sr.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 89969 entries, 0 to 89968
Data columns (total 13 columns):
tconst_x          89969 non-null object
titleType         89969 non-null object
primaryTitle      89969 non-null object
originalTitle     89969 non-null object
isAdult           89969 non-null int64
startYear         89969 non-null object
endYear           89969 non-null object
runtimeMinutes    89969 non-null object
genres            89969 non-null object
tconst_y          89969 non-null object
parentTconst      89969 non-null object
seasonNumber      89969 non-null object
episodeNumber     89969 non-null object
dtypes: int64(1), object(12)
memory usage: 9.6+ MB


In [109]:
m.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 452 entries, 0 to 451
Data columns (total 17 columns):
tconst             452 non-null object
titleType          452 non-null object
primaryTitle       452 non-null object
originalTitle      452 non-null object
isAdult            452 non-null int64
startYear          452 non-null object
endYear            452 non-null object
runtimeMinutes     452 non-null object
genres             452 non-null object
titleId            452 non-null object
ordering           452 non-null int64
title              452 non-null object
region             452 non-null object
language           452 non-null object
types              452 non-null object
attributes         452 non-null object
isOriginalTitle    452 non-null object
dtypes: int64(2), object(15)
memory usage: 63.6+ KB


In [111]:
m.titleType.unique()

array(['short', 'movie'], dtype=object)

In [112]:
m.titleType.value_counts()

short    449
movie      3
Name: titleType, dtype: int64

In [139]:
cast = pd.read_csv('title.principals.tsv',sep='\t')

In [140]:
cast.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32227696 entries, 0 to 32227695
Data columns (total 6 columns):
tconst        object
ordering      int64
nconst        object
category      object
job           object
characters    object
dtypes: int64(1), object(5)
memory usage: 1.4+ GB


In [149]:
series = pd.read_csv('title.episode.tsv',sep= '\t')

In [88]:
a = pd.DataFrame({'a':[1,2,3,4], 'h':[32,332,4,32]})

In [86]:
b = pd.DataFrame({'a':[1,2,3,4], 'x':[43,434,554,22]})

In [82]:
([a,b],axis = 1)

a    b  a    b
0  1   32  1   43
1  2  332  2  434
2  3    4  3  554
3  4   32  4   22

In [94]:
pd.merge(a,b,how='left',left_on='a', right_on='a')

a    h    x
0  1   32   43
1  2  332  434
2  3    4  554
3  4   32   22

In [189]:
titles = pd.read_csv('title.basics.tsv',sep='\t')
titles.drop(['originalTitle'],inplace = True,axis=1)

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [190]:
titles = titles.drop_duplicates(subset = ['primaryTitle','titleType','startYear'])

In [194]:
titles.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 3847051 entries, 0 to 5631728
Data columns (total 8 columns):
tconst            object
titleType         object
primaryTitle      object
isAdult           int64
startYear         object
endYear           object
runtimeMinutes    object
genres            object
dtypes: int64(1), object(7)
memory usage: 424.2+ MB


In [ ]:
principle = pd.read_csv('title.principals.tsv',sep='\t')

In [ ]:
principle.info()

In [1]:
import pandas as pd

In [2]:
principle = pd.read_csv('title.principals.tsv',sep='\t')

In [3]:
principle.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32227696 entries, 0 to 32227695
Data columns (total 6 columns):
tconst        object
ordering      int64
nconst        object
category      object
job           object
characters    object
dtypes: int64(1), object(5)
memory usage: 1.4+ GB


In [6]:
principle.category.unique()

array(['self', 'director', 'cinematographer', 'composer', 'producer',
       'editor', 'actor', 'actress', 'writer', 'production_designer',
       'archive_footage', 'archive_sound'], dtype=object)

In [9]:
principle.job.nunique()

30835

In [17]:
principle[principle['category']=='actress'].tail(100)

tconst  ordering      nconst category job  \
32227142  tt9758198         2   nm5274014  actress  \N   
32227147  tt9758200         1   nm5199119  actress  \N   
32227148  tt9758202         1   nm5199119  actress  \N   
32227149  tt9758206         1   nm5199119  actress  \N   
32227154  tt9758210         1   nm5199119  actress  \N   
32227158  tt9758214         1  nm10297770  actress  \N   
32227167  tt9758228         1   nm5199119  actress  \N   
32227168  tt9758232         1   nm5274014  actress  \N   
32227174  tt9758234         1   nm5199119  actress  \N   
32227176  tt9758238         1   nm5199119  actress  \N   
32227177  tt9758240         1   nm5199119  actress  \N   
32227178  tt9758240         2   nm3131329  actress  \N   
32227179  tt9758242         1   nm5199119  actress  \N   
32227195  tt9758250         1   nm5199119  actress  \N   
32227196  tt9758250         2   nm3131329  actress  \N   
32227198  tt9758252         1  nm10297770  actress  \N   
32227208  tt9758254         1   nm0209663  actress  \N   
32227209  tt9758254         2   nm0580158  actress  \N   
32227218  tt9758256         2   nm5274014  actress  \N   
32227223  tt9758258         1   nm5199119  actress  \N   
32227225  tt9758260         1   nm7036568  actress  \N   
32227226  tt9758260         2   nm4706392  actress  \N   
32227236  tt9758264         1   nm5199119  actress  \N   
32227238  tt9758266         1  nm10465098  actress  \N   
32227240  tt9758266         3  nm10465315  actress  \N   
32227242  tt9758266         5  nm10465304  actress  \N   
32227247  tt9758268         1   nm5199119  actress  \N   
32227252  tt9758272         2   nm5274014  actress  \N   
32227254  tt9758274         1   nm5199119  actress  \N   
32227255  tt9758276         1   nm5199119  actress  \N   
...             ...       ...         ...      ...  ..   
32227516  tt9758498         1  nm10465464  actress  \N   
32227518  tt9758498         3  nm10465461  actress  \N   
32227523  tt9758498         8  nm10465463  actress  \N   
32227526  tt9758504         2   nm0484929  actress  \N   
32227532  tt9758510         1   nm5421050  actress  \N   
32227542  tt9758516         1   nm0093698  actress  \N   
32227559  tt9758532         1   nm9460821  actress  \N   
32227571  tt9758540         3   nm1595011  actress  \N   
32227574  tt9758542         1   nm0523344  actress  \N   
32227575  tt9758542         2   nm0482941  actress  \N   
32227576  tt9758542         3   nm0226326  actress  \N   
32227582  tt9758542         9   nm7355195  actress  \N   
32227587  tt9758546         1   nm5879098  actress  \N   
32227590  tt9758546         4   nm4469341  actress  \N   
32227598  tt9758552         2   nm1008046  actress  \N   
32227599  tt9758552         3   nm6422413  actress  \N   
32227612  tt9758566         2   nm1008046  actress  \N   
32227613  tt9758566         3   nm6422413  actress  \N   
32227621  tt9758574         1   nm6814486  actress  \N   
32227623  tt9758574         3  nm10465584  actress  \N   
32227626  tt9758574         6   nm6581285  actress  \N   
32227628  tt9758574         8  nm10465583  actress  \N   
32227632  tt9758590         2   nm1008046  actress  \N   
32227633  tt9758590         3   nm6422413  actress  \N   
32227640  tt9758594         1  nm10270605  actress  \N   
32227641  tt9758594         2   nm8397988  actress  \N   
32227652  tt9758614         2   nm1008046  actress  \N   
32227653  tt9758614         3   nm6422413  actress  \N   
32227672  tt9758618         2   nm1008046  actress  \N   
32227673  tt9758618         3   nm6422413  actress  \N   

                               characters  
32227142                               \N  
32227147                               \N  
32227148                               \N  
32227149                               \N  
32227154                               \N  
32227158  ["Fondatrice de Shauna Events"]  
32227167                               \N  
32227168                               \N  
32227174            

In [18]:
import pandas as pd 

In [19]:
title = pd.read_csv('title.basics.tsv',sep = '\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [20]:
title.info ()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5631729 entries, 0 to 5631728
Data columns (total 9 columns):
tconst            object
titleType         object
primaryTitle      object
originalTitle     object
isAdult           int64
startYear         object
endYear           object
runtimeMinutes    object
genres            object
dtypes: int64(1), object(8)
memory usage: 386.7+ MB


In [29]:
title = title[['tconst','titleType', 'primaryTitle', 'isAdult', 'startYear', 'genres']]

In [30]:
title.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5631729 entries, 0 to 5631728
Data columns (total 6 columns):
tconst          object
titleType       object
primaryTitle    object
isAdult         int64
startYear       object
genres          object
dtypes: int64(1), object(5)
memory usage: 257.8+ MB


In [67]:
title.head()

tconst titleType            primaryTitle  isAdult startYear  \
0  tt0000001     short              Carmencita        0      1894   
1  tt0000002     short  Le clown et ses chiens        0      1892   
2  tt0000003     short          Pauvre Pierrot        0      1892   
3  tt0000004     short             Un bon bock        0      1892   
4  tt0000005     short        Blacksmith Scene        0      1893   

                     genres                                newCol  
0         Documentary,Short  Insert into title values( 'short', '  
1           Animation,Short  Insert into title values( 'short', '  
2  Animation,Comedy,Romance  Insert into title values( 'short', '  
3           Animation,Short  Insert into title values( 'short', '  
4              Comedy,Short  Insert into title values( 'short', '

In [64]:
title['newCol'] = title.apply(lambda row: "Insert into title values( '" + row['titleType'] + "', '", axis = 1 )

In [ ]:
import pandas as pd 

title = title[['tconst','titleType', 'primaryTitle', 'isAdult', 'startYear', 'genres']]


In [106]:
title = pd.read_csv('title.basics.tsv',sep = '\t')

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [107]:
title.head() 

tconst titleType            primaryTitle           originalTitle  \
0  tt0000001     short              Carmencita              Carmencita   
1  tt0000002     short  Le clown et ses chiens  Le clown et ses chiens   
2  tt0000003     short          Pauvre Pierrot          Pauvre Pierrot   
3  tt0000004     short             Un bon bock             Un bon bock   
4  tt0000005     short        Blacksmith Scene        Blacksmith Scene   

   isAdult startYear endYear runtimeMinutes                    genres  
0        0      1894      \N              1         Documentary,Short  
1        0      1892      \N              5           Animation,Short  
2        0      1892      \N              4  Animation,Comedy,Romance  
3        0      1892      \N             \N           Animation,Short  
4        0      1893      \N              1              Comedy,Short

In [112]:
title = title.sample(1000)

/Users/ketankokane/anaconda3/lib/python3.6/site-packages/IPython/core/interactiveshell.py:2698: DtypeWarning: Columns (5) have mixed types. Specify dtype option on import or set low_memory=False.
  interactivity=interactivity, compiler=compiler, result=result)


In [123]:
data.columns

Index(['tconst', 'titleType', 'primaryTitle', 'originalTitle', 'isAdult',
       'startYear', 'endYear', 'runtimeMinutes', 'genres'],
      dtype='object')

In [449]:
TITLE_TSV_COLUMNS = ['tconst','titleType', 'primaryTitle', 'isAdult', 'releaseYear', 'endYear','runtimeMinutes','genres']
PRIMARY_KEY_COLUMNS = ['titleType', 'primaryTitle']
TITLE_COLUMN_NAME = ['titleType', 'primaryTitle', 'releaseYear' ,'isAdult'] #correct
TITLE_GENRES = ['titleType', 'primaryTitle','genres'] # correct 
TITLE_RATINGS = PRIMARY_KEY_COLUMNS + ['averageRating','numVotes'] #correct 
MOVIE_COLUMN_NAME = ['titleType', 'primaryTitle', 'runtimeMinutes'] #correct 
VIDEOGAME_COLUMN_NAME = ['titleType', 'primaryTitle' ,'runtimeMinutes'] # correct 
TV_EPISODE_COLUMN_NAME = ['titleType', 'primaryTitle', 'runtimeMinutes'] # correct
TV_SERIES_COLUMN_NAME = ['titleType', 'primaryTitle', 'endYear','runtimeMinutes'] # correct

file = open('sql/populate_all.sql','w') 
file.close()
data = pd.read_csv('title.basics.tsv',sep='\t')
data = data.sample(10000)
temp = data[['tconst','titleType', 'primaryTitle', 'isAdult', 'startYear', 'endYear','runtimeMinutes','genres']]
temp.dropna(inplace=True)
temp = temp.drop_duplicates(subset=PRIMARY_KEY_COLUMNS)
temp.columns = TITLE_TSV_COLUMNS
#create a duplicate title.basic with just 10000 shuffled entries
temp.to_csv('master_titles.tsv',sep='\t',index=False)
master = temp
#drop the duplciates
temp = temp.drop_duplicates(subset=PRIMARY_KEY_COLUMNS)
# generate the title table
temp = temp[['tconst']+ TITLE_COLUMN_NAME]
temp = temp.drop_duplicates(subset=PRIMARY_KEY_COLUMNS)
generateSqlAndTsvFile(temp,TITLE_COLUMN_NAME, 'title' )

# generate ratings table
#data = pd.read_csv('title_ketan.tsv',sep='\t') # master data only
temp = master
temp = temp[['tconst']+PRIMARY_KEY_COLUMNS]
temp = temp.drop_duplicates(subset=PRIMARY_KEY_COLUMNS)

ratings = pd.read_csv('title.ratings.tsv',sep='\t')
temp = temp.merge(ratings,how = 'inner', on='tconst')
generateSqlAndTsvFile(temp,TITLE_RATINGS, 'titleRatings')

#genres 
temp = master
temp = temp[['tconst']+TITLE_GENRES]
temp = temp.drop_duplicates(subset=PRIMARY_KEY_COLUMNS)

temp = pd.DataFrame(temp.genres.str.split(',').tolist(), index=[temp.tconst,temp.titleType,temp.primaryTitle]).stack()
temp = temp.reset_index()[['tconst','titleType','primaryTitle',0]]
temp.columns = [['tconst','titleType','primaryTitle','genres']]
temp = temp.drop_duplicates(subset=PRIMARY_KEY_COLUMNS+['genres'])
generateSqlAndTsvFile(temp,TITLE_GENRES, 'titleGenres')

temp = master
temp = temp[temp.titleType.isin(['tvEpisode'])]
temp = temp[['tconst'] + TV_EPISODE_COLUMN_NAME ]
temp = temp.drop_duplicates(subset=PRIMARY_KEY_COLUMNS)
#TV_EPISODE_COLUMN_NAME = ['titleType', 'primaryTitle','releaseYear', 'runtimeMinutes'] # correct
temp = temp.drop_duplicates(subset=TV_EPISODE_COLUMN_NAME)

generateSqlAndTsvFile(temp,TV_EPISODE_COLUMN_NAME, 'tvEpisode')

temp = master
temp = temp[temp.titleType.isin(['videoGame'])]
temp = temp[['tconst'] + VIDEOGAME_COLUMN_NAME]
temp = temp.drop_duplicates(subset=PRIMARY_KEY_COLUMNS)
#TV_EPISODE_COLUMN_NAME = ['titleType', 'primaryTitle','releaseYear', 'runtimeMinutes'] # correct
temp = temp.drop_duplicates(subset=VIDEOGAME_COLUMN_NAME)

generateSqlAndTsvFile(temp,VIDEOGAME_COLUMN_NAME, 'videoGame' )

temp = master
temp = temp[~temp.titleType.isin(['tvEpisode','videoGame','tvSeries','tvMiniSeries','video' ])]
temp = temp[['tconst'] + MOVIE_COLUMN_NAME]
temp = temp.drop_duplicates(subset=PRIMARY_KEY_COLUMNS)
#TV_EPISODE_COLUMN_NAME = ['titleType', 'primaryTitle','releaseYear', 'runtimeMinutes'] # correct
temp = temp.drop_duplicates(subset=MOVIE_COLUMN_NAME)
generateSqlAndTsvFile(temp,MOVIE_COLUMN_NAME, 'movie' )

temp = master
temp = temp[temp.titleType.isin(['tvSeries','tvMiniSeries'])]
temp = temp[['tconst'] + TV_SERIES_COLUMN_NAME]
temp = temp.drop_duplicates(subset=PRIMARY_KEY_COLUMNS)
temp = temp.drop_duplicates(subset=TV_SERIES_COLUMN_NAME)
generateSqlAndTsvFile(temp,TV_SERIES_COLUMN_NAME, 'tvseries' )

tconst = master[['tconst']]
person = pd.read_csv('title.principals.tsv',sep='\t')
PERSON_PRIMAY_KEY = ['primaryName','birthYear']
p = person[['tconst','nconst']]
x = tconst.merge(p,how = 'inner', on = 'tconst')
data1 = x.merge(person, how='inner', on = ['tconst','nconst'])
data1.drop(['ordering'],axis = 1, inplace=True)
names = pd.read_csv('name.basics.tsv',sep='\t')
names.drop(['primaryProfession','knownForTitles'],inplace = True, axis = 1)
names = names.drop_duplicates(subset = PERSON_PRIMAY_KEY)
t = data1.merge(names, how = 'inner', on = 'nconst') 
t.to_csv('principle_cast.tsv',sep='\t')

person = t 
PERSON_COLUMN_NAME = ['primaryName','birthYear','deathYear']
person = person[['tconst','nconst']+PERSON_COLUMN_NAME]
person = person.drop_duplicates(subset=PERSON_PRIMAY_KEY)
generateSqlAndTsvFile(person,PERSON_COLUMN_NAME, 'person')
#fine

temp = t
temp = temp[~temp.category.isin(['actor','director','writer','self','actress'])]
MISC_COLUMN_NAME = PERSON_PRIMAY_KEY + ['category'] 
temp = temp[['tconst','nconst'] + MISC_COLUMN_NAME  + ['job']]
temp = temp.drop_duplicates(subset=MISC_COLUMN_NAME)
generateSqlAndTsvFile(temp,MISC_COLUMN_NAME,'credit')


temp = t
temp = temp[temp.category.isin(['actor','self','actress'])]
ACTOR_COLUMN_NAME = PERSON_PRIMAY_KEY + ['category'] 
temp = temp[['tconst','nconst'] + ACTOR_COLUMN_NAME  + ['characters']]
temp = temp.drop_duplicates(subset=ACTOR_COLUMN_NAME)
generateSqlAndTsvFile(temp,ACTOR_COLUMN_NAME,'actor')


temp = t
temp = temp[temp.category.isin(['director'])]
DIRECTOR_COLUMN_NAME = PERSON_PRIMAY_KEY + ['job']
temp = temp[['tconst','nconst'] + DIRECTOR_COLUMN_NAME] 
temp = temp.drop_duplicates(subset=DIRECTOR_COLUMN_NAME)
generateSqlAndTsvFile(temp,DIRECTOR_COLUMN_NAME,'director')


temp = t
temp = temp[temp.category.isin(['writer'])]
WRITER_COLUMN_NAME = PERSON_PRIMAY_KEY + ['category']
temp = temp[['tconst','nconst'] + WRITER_COLUMN_NAME +['job']]
temp = temp.drop_duplicates(subset=WRITER_COLUMN_NAME)
generateSqlAndTsvFile(temp,WRITER_COLUMN_NAME,'writer')


title = pd.read_csv('csv/title_ketan.tsv', sep= '\t')
writer = pd.read_csv('csv/writer_ketan.tsv',sep= '\t')
title_writer = title.merge(writer, how='inner', on= 'tconst')
TITLE_WRITER_COLUMN_NAMES = ['titleType', 'primaryTitle','primaryName', 'birthYear','job']
title_writer = title_writer [TITLE_WRITER_COLUMN_NAMES]
title_writer = title_writer.drop_duplicates(subset=TITLE_WRITER_COLUMN_NAMES)
generateSqlAndTsvFile(title_writer, TITLE_WRITER_COLUMN_NAMES, 'WritesFor')

director = pd.read_csv('csv/director_ketan.tsv',sep= '\t')
title_director = title.merge(director, how='inner', on= 'tconst')
TITLE_DIRECTOR_COLUMN_NAMES = ['titleType', 'primaryTitle', 'primaryName', 'birthYear', 'job']
title_director = title_director [TITLE_DIRECTOR_COLUMN_NAMES]
title_director = title_director.drop_duplicates(subset=TITLE_DIRECTOR_COLUMN_NAMES)
generateSqlAndTsvFile(title_director, TITLE_DIRECTOR_COLUMN_NAMES, 'title_director')

title_credit = pd.read_csv('csv/credit_ketan.tsv',sep= '\t')
title_credit = title.merge(title_credit, how='inner', on= 'tconst')
TITLE_CREDIT_COLUMN_NAMES = ['titleType', 'primaryTitle', 'primaryName', 'birthYear', 'category', 'job']
title_credit = title_credit [TITLE_CREDIT_COLUMN_NAMES]
title_credit = title_credit.drop_duplicates(subset=TITLE_CREDIT_COLUMN_NAMES)
generateSqlAndTsvFile(title_credit, TITLE_CREDIT_COLUMN_NAMES, 'title_credit')

actor = pd.read_csv('csv/actor_ketan.tsv',sep= '\t')
title_actor = title.merge(actor, how='inner', on= 'tconst')
TITLE_ACTOR_COLUMN_NAMES = ['titleType', 'primaryTitle', 'primaryName', 'birthYear', 'category','characters']
title_actor = title_actor [TITLE_ACTOR_COLUMN_NAMES]
title_actor = title_actor.drop_duplicates(subset=TITLE_ACTOR_COLUMN_NAMES)
generateSqlAndTsvFile(title_actor, TITLE_ACTOR_COLUMN_NAMES, 'title_actor')


tvepisode = pd.read_csv('csv/tvEpisode_ketan.tsv',sep='\t')

#tvepisode = tvepisode[['tconst'] + PRIMARY_KEY_COLUMNS]

tvseries = pd.read_csv('csv/tvseries_ketan.tsv',sep='\t')

#tvseries = tvseries[['tconst'] + PRIMARY_KEY_COLUMNS]

e = pd.read_csv('title.episode.tsv',sep='\t')

tve = tvepisode.merge(e , how = 'inner', on = 'tconst')

tvee = tve.merge(tvseries, left_on='parentTconst', right_on='tconst', how='inner')
tvee.drop(['tconst_x','tconst_y','parentTconst'],axis = 1, inplace = True)

tvee.columns = ['episodeTitleType', 'episodePrimaryTitle', 'episodeReleaseYear',
       'seasonNumber', 'episodeNumber', 'seriesTitleType', 'seriesPrimaryTitle',
       'endYear','releaseYear']

generateSqlAndTsvFile(tvee,['episodeTitleType', 'episodePrimaryTitle', 'episodeReleaseYear',
       'seasonNumber', 'episodeNumber', 'seriesTitleType', 'seriesPrimaryTitle',
       'endYear','releaseYear'],'belongs')

In [486]:
tvee.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 24 entries, 0 to 23
Data columns (total 9 columns):
episodeTitleType       24 non-null object
episodePrimaryTitle    24 non-null object
episodeReleaseYear     24 non-null object
seasonNumber           24 non-null object
episodeNumber          24 non-null object
seriesTitleType        24 non-null object
seriesPrimaryTitle     24 non-null object
endYear                24 non-null object
releaseYear            24 non-null object
dtypes: object(9)
memory usage: 1.9+ KB


In [468]:
tvepisode = pd.read_csv('csv/tvEpisode_ketan.tsv',sep='\t')

#tvepisode = tvepisode[['tconst'] + PRIMARY_KEY_COLUMNS]

tvseries = pd.read_csv('csv/tvseries_ketan.tsv',sep='\t')

#tvseries = tvseries[['tconst'] + PRIMARY_KEY_COLUMNS]

e = pd.read_csv('title.episode.tsv',sep='\t')

tve = tvepisode.merge(e , how = 'inner', on = 'tconst')

tvee = tve.merge(tvseries, left_on='parentTconst', right_on='tconst', how='inner')
tvee.drop(['tconst_x','tconst_y','parentTconst'],axis = 1, inplace = True)

tvee.columns = ['episodeTitleType', 'episodePrimaryTitle', 'episodeReleaseYear',
       'seasonNumber', 'episodeNumber', 'seriesTitleType', 'seriesPrimaryTitle',
       'endYear','releaseYear']

generateSqlAndTsvFile(tvee,['episodeTitleType', 'episodePrimaryTitle', 'episodeReleaseYear',
       'seasonNumber', 'episodeNumber', 'seriesTitleType', 'seriesPrimaryTitle',
       'endYear','releaseYear'],'belongs')

In [469]:
tvee.head()

episodeTitleType                               episodePrimaryTitle  \
0        tvEpisode                            Unconscious Conscience   
1        tvEpisode                                   Advantage Shift   
2        tvEpisode  Sanibel Island, Palm Tree, White Sand Beach View   
3        tvEpisode                                    Til Vesterålen   
4        tvEpisode                        Episode dated 30 July 2012   

  episodeReleaseYear seasonNumber episodeNumber seriesTitleType  \
0                 \N            1            21        tvSeries   
1                 \N            2             4        tvSeries   
2                 \N            2            12        tvSeries   
3                 \N            5             1        tvSeries   
4                 \N           \N            \N        tvSeries   

                 seriesPrimaryTitle endYear releaseYear  
0  Lippy the Lion and Hardy Har Har      \N          30  
1                         War: 2600      \N          \N  
2      What's for Sale? With a View      \N          22  
3                      Gutta på tur      \N          50  
4                   CP24 Live at 10      \N          \N

In [456]:
tvee.info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 9 columns):
episodeTitleType       0 non-null object
episodePrimaryTitle    0 non-null object
episodeReleaseYear     0 non-null object
seasonNumber           0 non-null object
episodeNumber          0 non-null object
seriesTitleType        0 non-null object
seriesPrimaryTitle     0 non-null object
endYear                0 non-null object
releaseYear            0 non-null object
dtypes: object(9)
memory usage: 0.0+ bytes


In [444]:
tvee.drop(['tconst_x','tconst_y','parentTconst'],axis = 1, inplace = True)

tvee.columns = ['episodeTitleType', 'episodePrimaryTitle', 'episodeReleaseYear',
       'seasonNumber', 'episodeNumber', 'seriesTitleType', 'seriesPrimaryTitle',
       'endYear','releaseYear']

In [437]:
tve.head()

tconst  titleType       primaryTitle runtimeMinutes parentTconst  \
0  tt0898108  tvEpisode  An Eye for an Eye             \N    tt0159898   
1  tt8008092  tvEpisode     Episode #1.223             \N    tt0398457   
2  tt1392543  tvEpisode       Episode #1.6             \N    tt0066643   
3  tt3511474  tvEpisode      Burning Woman             43    tt2402061   
4  tt9753530  tvEpisode      Episode #1.42             \N    tt6131658   

  seasonNumber episodeNumber  
0            6             7  
1            1           223  
2            1             6  
3            1             4  
4            1            42

In [440]:
tve.head()

tconst  titleType       primaryTitle runtimeMinutes parentTconst  \
0  tt0898108  tvEpisode  An Eye for an Eye             \N    tt0159898   
1  tt8008092  tvEpisode     Episode #1.223             \N    tt0398457   
2  tt1392543  tvEpisode       Episode #1.6             \N    tt0066643   
3  tt3511474  tvEpisode      Burning Woman             43    tt2402061   
4  tt9753530  tvEpisode      Episode #1.42             \N    tt6131658   

  seasonNumber episodeNumber  
0            6             7  
1            1           223  
2            1             6  
3            1             4  
4            1            42

In [441]:
tvseries.head()

tconst titleType      primaryTitle endYear runtimeMinutes
0  tt1160470  tvSeries    Miljoenenjacht      \N             \N
1  tt6625986  tvSeries         The Guest      \N             \N
2  tt8268814  tvSeries  Rock Fashion Art      \N             \N
3  tt3264604  tvSeries   Generation Cryo      \N             \N
4  tt3002378  tvSeries        Go Figure!      \N             \N

In [314]:
TITLE_TSV_COLUMNS = ['tconst','titleType', 'primaryTitle', 'isAdult', 'releaseYear', 'endYear','runtimeMinutes','genres']
PRIMARY_KEY_COLUMNS = ['titleType', 'primaryTitle','releaseYear']
TITLE_COLUMN_NAME = ['titleType', 'primaryTitle', 'releaseYear' ,'isAdult'] #correct
TITLE_GENRES = ['titleType', 'primaryTitle', 'releaseYear','genres'] # correct 
TITLE_RATINGS = PRIMARY_KEY_COLUMNS + ['averageRating','numVotes'] #correct 
MOVIE_COLUMN_NAME = ['titleType', 'primaryTitle','releaseYear', 'runtimeMinutes'] #correct 
VIDEOGAME_COLUMN_NAME = ['titleType', 'primaryTitle','releaseYear', 'runtimeMinutes'] # correct 
TV_EPISODE_COLUMN_NAME = ['titleType', 'primaryTitle','releaseYear', 'runtimeMinutes'] # correct
TV_SERIES_COLUMN_NAME = ['titleType', 'primaryTitle','releaseYear', 'endYear','runtimeMinutes'] # correct

In [348]:
tvepisode.head()

tconst  titleType                   primaryTitle releaseYear  \
0  tt5479268  tvEpisode   Game of the Year: 420BlazeIt        2015   
1  tt1044223  tvEpisode      Episode dated 24 May 2007        2007   
2  tt4391542  tvEpisode  Episode dated 4 February 2015        2015   
3  tt0868676  tvEpisode                 Colours: Green        1998   
4  tt1874012  tvEpisode  Episode dated 22 January 2000        2000   

  runtimeMinutes  
0             \N  
1             \N  
2             \N  
3             \N  
4             \N

In [349]:
tvseries.head()

tconst     titleType              primaryTitle releaseYear endYear  \
0  tt1582661      tvSeries               8 ston aera        1999      \N   
1  tt0078685      tvSeries             Seeing Things        1981    1987   
2  tt3856314      tvSeries            Golden Barista        2014      \N   
3  tt0163507      tvSeries  Whose Line Is It Anyway?        1998    2007   
4  tt9510012  tvMiniSeries     Melodifestivalen 2019        2019    2019   

  runtimeMinutes  
0             \N  
1             \N  
2             \N  
3             30  
4             \N

In [350]:
tvsepisodebelongstvseries = tvepisode.merge(tvseries,how='inner',on='tconst')

In [351]:
tvsepisodebelongstvseries.info()

<class 'pandas.core.frame.DataFrame'>
Index: 0 entries
Data columns (total 10 columns):
tconst              0 non-null object
titleType_x         0 non-null object
primaryTitle_x      0 non-null object
releaseYear_x       0 non-null object
runtimeMinutes_x    0 non-null object
titleType_y         0 non-null object
primaryTitle_y      0 non-null object
releaseYear_y       0 non-null object
endYear             0 non-null object
runtimeMinutes_y    0 non-null object
dtypes: object(10)
memory usage: 0.0+ bytes


In [302]:
title_writer.head()

titleType                        primaryTitle releaseYear      primaryName  \
0  tvEpisode  Guess That Celebrity Tattoo (GAME)        2017   Lizzie Bassett   
1  tvEpisode  Guess That Celebrity Tattoo (GAME)        2017   Edward Coleman   
2  tvEpisode  Guess That Celebrity Tattoo (GAME)        2017     Micah Gordon   
3  tvEpisode  Guess That Celebrity Tattoo (GAME)        2017  Kevin Kostelnik   
4  tvEpisode  Guess That Celebrity Tattoo (GAME)        2017   Ellie McElvain   

  birthYear job  
0        \N  \N  
1        \N  \N  
2        \N  \N  
3        \N  \N  
4        \N  \N

In [285]:
temp = temp[['tconst','nconst'] + DIRECTOR_COLUMN_NAME]

In [290]:
temp.head(100)

tconst     nconst category                 job characters  \
18   tt7557372  nm5668420   writer                  \N         \N   
19   tt7557372  nm6817686   writer                  \N         \N   
20   tt7557372  nm4880566   writer                  \N         \N   
21   tt7557372  nm6212318   writer                  \N         \N   
22   tt7557372  nm8203568   writer                  \N         \N   
33   tt1345825  nm1346477   writer          screenplay         \N   
52   tt4953866  nm0374289   writer          written by         \N   
53   tt4953866  nm2360862   writer         head writer         \N   
75   tt0485777  nm2088941   writer                  \N         \N   
85   tt0754847  nm0656700   writer             creator         \N   
86   tt0754847  nm0656749   writer             creator         \N   
101  tt7516994  nm7677494   writer                  \N         \N   
143  tt9743586  nm7189902   writer             creator         \N   
149  tt9743586  nm4673844   writer                  \N         \N   
150  tt9743586  nm8687418   writer                  \N         \N   
151  tt9743586  nm5681364   writer                  \N         \N   
152  tt9743586  nm7129304   writer                  \N         \N   
157  tt7016262  nm0853112   writer    special material         \N   
189  tt1863872  nm1038302   writer             creator         \N   
190  tt1863872  nm0141139   writer            dialogue         \N   
191  tt1863872  nm1839148   writer   story coordinator         \N   
205  tt9515080  nm8513709   writer              writer         \N   
216  tt0690822  nm1115314   writer          written by         \N   
222  tt0690822  nm0542425   writer          written by         \N   
223  tt0690822  nm0612306   writer          written by         \N   
224  tt0690822  nm1618230   writer          written by         \N   
225  tt0690822  nm0710898   writer          written by         \N   
232  tt8403072  nm0722865   writer              writer         \N   
233  tt8859810  nm0722865   writer                  \N         \N   
246  tt7731888  nm1369367   writer        screenwriter         \N   
..         ...        ...      ...                 ...        ...   
661  tt4319818  nm0229890   writer             creator         \N   
662  tt4510290  nm0229890   writer             creator         \N   
663  tt4319818  nm0229908   writer             creator         \N   
664  tt4510290  nm0229908   writer             creator         \N   
665  tt4319818  nm0040612   writer              writer         \N   
666  tt4319818  nm0349953   writer              writer         \N   
678  tt2779430  nm1386702   writer          adaptation         \N   
685  tt2779430  nm1039199   writer       co-adaptation         \N   
686  tt2779430  nm1185004   writer       co-adaptation         \N   
687  tt2779430  nm2336044   writer      original story         \N   
693  tt0927694  nm0475001   writer              writer         \N   
694  tt0927694  nm0475003   writer              writer         \N   
725  tt7695860  nm1888485   writer              script         \N   
732  tt7695860  nm1354318   writer          adaptation         \N   
733  tt7695860  nm1898552   writer              script         \N   
749  tt6338496  nm8386754   writer                  \N         \N   
757  tt6586640  nm0327964   writer                  \N         \N   
762  tt0536748  nm0201751   writer             creator         \N   
763  tt0536761  nm0201751   writer             creator         \N   
769  tt0591700  nm0521143   writer             creator         \N   
770  tt0591700  nm0807482   writer          written by         \N   
791  tt2705560  nm5593961   writer          screenplay         \N   
792  tt2705560  nm2401269   writer  assistant dialogue         \N   
811  tt8063512  nm7663869   writer                  \N         \N   
836  tt1246134  nm0908194   writer          written by         \N   
849  tt2078693  nm2120003   writer                book         \N   
850  t

In [260]:
person.head()

tconst     nconst        primaryName birthYear deathYear
0  tt6600304  nm5225923       Eric Hinwood        \N        \N
1  tt6600304  nm1487328        X. Dean Lim        \N        \N
2  tt6600304  nm4707279  Jennifer Chandler        \N        \N
3  tt6600304  nm0728257        Fred Risher        \N        \N
4  tt6600304  nm3440958  Casey Christopher      1984        \N

In [246]:
director = data1[data1['category'] == 'director']

In [247]:
director.head()

tconst     nconst  category              job characters
1   tt6600304  nm1487328  director               \N         \N
9   tt8293894  nm3961568  director               \N         \N
10  tt8293894  nm4342517  director               \N         \N
11  tt8293894  nm1785633  director  series director         \N
32  tt1345825  nm1033529  director               \N         \N

In [245]:
data1.head(100)

tconst      nconst         category                      job  \
0   tt6600304   nm5225923            actor                       \N   
1   tt6600304   nm1487328         director                       \N   
2   tt6600304   nm4707279         producer                 producer   
3   tt6600304   nm0728257         producer                 producer   
4   tt6600304   nm3440958           editor                       \N   
5   tt6600304   nm7038595           editor                       \N   
6   tt6600304   nm8210847           editor                       \N   
7   tt6600304   nm7741549           editor                       \N   
8   tt8293894   nm2763922             self                       \N   
9   tt8293894   nm3961568         director                       \N   
10  tt8293894   nm4342517         director                       \N   
11  tt8293894   nm1785633         director          series director   
12  tt8293894   nm3204286         composer                       \N   
13  tt8293894   nm3606960  cinematographer  director of photography   
14  tt8293894   nm0055285           editor          off-line editor   
15  tt8293894   nm4869360           editor          off-line editor   
16  tt7557372   nm2561248             self                       \N   
17  tt7557372   nm0623641             self                       \N   
18  tt7557372   nm5668420           writer                       \N   
19  tt7557372   nm6817686           writer                       \N   
20  tt7557372   nm4880566           writer                       \N   
21  tt7557372   nm6212318           writer                       \N   
22  tt7557372   nm8203568           writer                       \N   
23  tt7557372   nm1701470           editor                       \N   
24  tt6790578   nm2360844            actor                       \N   
25  tt6790578   nm6135332            actor                       \N   
26  tt6790578   nm8927442          actress                       \N   
27  tt6790578   nm8928405          actress                       \N   
28  tt1345825   nm0263594            actor                       \N   
29  tt1345825   nm3254646          actress                       \N   
..        ...         ...              ...                      ...   
70  tt1123772   nm0802733           editor                       \N   
71  tt0485777   nm0624516             self                       \N   
72  tt0485777   nm6327076             self                       \N   
73  tt0485777   nm6327079             self                       \N   
74  tt0485777   nm6327077             self                       \N   
75  tt0485777   nm2088941           writer                       \N   
76  tt0485777   nm6327080             self                       \N   
77  tt0485777   nm6327075             self                       \N   
78  tt0485777   nm6327078             self                       \N   
79  tt0485777   nm0386944  archive_footage                       \N   
80  tt0754847   nm0004757          actress                       \N   
81  tt0754847   nm0742146            actor                       \N   
82  tt0754847   nm0005439          actress                       \N   
83  tt0754847   nm0608378            actor                       \N   
84  tt0754847   nm0957197         director                       \N   
85  tt0754847   nm0656700           writer                  creator   
86  tt0754847   nm0656749           writer                  creator   
87  tt0754847   nm0755254         producer                 producer   
88  tt0754847   nm0101663         composer                       \N   
89  tt9325462   nm2954746             self                       \N   
90  tt9325462   nm0098517             self                       \N   
91  tt9325462  nm10277294            actor                       \N   
92  tt9325462   nm0482914             self                       \N   
93  tt9325462   nm1303569             self                       \N   
94  tt3789740   nm5595697      

In [233]:
x.info()



person.head()

data1.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 5658 entries, 0 to 5657
Data columns (total 2 columns):
tconst    5658 non-null object
nconst    5658 non-null object
dtypes: object(2)
memory usage: 132.6+ KB


In [241]:
data1.head(100)

tconst      nconst  ordering         category                      job  \
0   tt6600304   nm5225923         1            actor                       \N   
1   tt6600304   nm1487328         2         director                       \N   
2   tt6600304   nm4707279         3         producer                 producer   
3   tt6600304   nm0728257         4         producer                 producer   
4   tt6600304   nm3440958         5           editor                       \N   
5   tt6600304   nm7038595         6           editor                       \N   
6   tt6600304   nm8210847         7           editor                       \N   
7   tt6600304   nm7741549         8           editor                       \N   
8   tt8293894   nm2763922         1             self                       \N   
9   tt8293894   nm3961568         2         director                       \N   
10  tt8293894   nm4342517         3         director                       \N   
11  tt8293894   nm1785633         4         director          series director   
12  tt8293894   nm3204286         5         composer                       \N   
13  tt8293894   nm3606960         6  cinematographer  director of photography   
14  tt8293894   nm0055285         7           editor          off-line editor   
15  tt8293894   nm4869360         8           editor          off-line editor   
16  tt7557372   nm2561248         1             self                       \N   
17  tt7557372   nm0623641         2             self                       \N   
18  tt7557372   nm5668420         3           writer                       \N   
19  tt7557372   nm6817686         4           writer                       \N   
20  tt7557372   nm4880566         5           writer                       \N   
21  tt7557372   nm6212318         6           writer                       \N   
22  tt7557372   nm8203568         7           writer                       \N   
23  tt7557372   nm1701470         8           editor                       \N   
24  tt6790578   nm2360844         1            actor                       \N   
25  tt6790578   nm6135332         2            actor                       \N   
26  tt6790578   nm8927442         3          actress                       \N   
27  tt6790578   nm8928405         4          actress                       \N   
28  tt1345825   nm0263594         1            actor                       \N   
29  tt1345825   nm3254646         2          actress                       \N   
..        ...         ...       ...              ...                      ...   
70  tt1123772   nm0802733         5           editor                       \N   
71  tt0485777   nm0624516         1             self                       \N   
72  tt0485777   nm6327076         2             self                       \N   
73  tt0485777   nm6327079         3             self                       \N   
74  tt0485777   nm6327077         4             self                       \N   
75  tt0485777   nm2088941         5           writer                       \N   
76  tt0485777   nm6327080         6             self                       \N   
77  tt0485777   nm6327075         7             self                       \N   
78  tt0485777   nm6327078         8             self                       \N   
79  tt0485777   nm0386944         9  archive_footage                       \N   
80  tt0754847   nm0004757         1          actress                       \N   
81  tt0754847   nm0742146         2            actor                       \N   
82  tt0754847   nm0005439         3          actress                       \N   
83  tt0754847   nm0608378         4            actor                       \N   
84  tt0754847   nm0957197         5         director                       \N   
85  tt0754847   nm0656700         6           writer                  creator   
86  tt0754847   nm0656749         7           writer                  creator   
87  tt0754847   nm0755254         8   

In [474]:
file = open('sql/new.sql.sql','w')
file.close()

In [479]:
def generateSqlAndTsvFile(data, columns, file_name ):
    file = open('sql/populate_all.sql','a') 
    # generate title table
    for index, row in data.iterrows():
        string = "Insert into "+ file_name + " values( "
        for col in columns:
            if type(row[col]) == str:
                if row[col] == '\\N':
                    string += "'NULL'" +", "
                else:
                    x = row[col].replace("'","''")
                    string += "'" + x + "', "
            else:
                string += "" + str(row[col]) +", "
        string = string[:-2]
        string += " ); \n"
        file.writelines(string)
    file.close() 
    data.to_csv('csv/' +file_name+ '_ketan.tsv', sep='\t',index = False)

In [326]:
"ketan's".replace("'","''")

"ketan''s"

In [152]:
data.titleType.unique()

array(['tvEpisode', 'tvMovie', 'movie', 'tvSeries', 'short', 'video',
       'videoGame', 'tvSpecial', 'tvShort', 'tvMiniSeries'], dtype=object)

In [151]:
data = pd.read_csv('title_ketan.tsv',sep='\t')

In [214]:
def generateSqlAndTsvFile(data, columns, file_name ):
    file = open('sql/'+ file_name+'.sql','w') 
    file.close()
    file = open('sql/'+ file_name+'.sql','a') 
    # generate title table
    for index, row in data.iterrows():
        string = "Insert into "+ file_name + " values( "
        for col in columns:
            if row[col] == '\\N' or row[col] == None:
                string += 'NULL' +", "
            if type(row[col]) != str:
                string += "" + str(row[col]) +", "
            else:
                string += "'" + row[col] + "', "

        string = string[:-2]
        string += " ); \n"
        file.writelines(string)
    file.close() 
    data.to_csv('csv/' +file_name+ '_ketan.tsv', sep='\t',index = False)

In [139]:
# make movie table
movie = pd.read_csv('title_ketan.tsv',sep='\t')
movie = movie[~ movie.titleType.isin(['tvEpisode','videoGame','tvSeries','tvMiniSeries','video' ])]

file = open('movie.sql','w') 
file.close()
file = open('movie.sql','a') 
# generate title table
for index, row in movie.iterrows():
    string = "Insert into movie values( "
    for col in MOVIE_COLUMN_NAME:
        if row[col] == '\\N' or row[col] == None:
            string += 'NULL' +", "
        if type(row[col]) != str:
            string += "" + str(row[col]) +", "
        else:
            string += "'" + row[col] + "', "
    
    string = string[:-2]
    string += " ); \n"
    file.writelines(string)
file.close() 
movie.to_csv('movie_ketan.tsv', sep='\t')

SyntaxError: (unicode error) 'unicodeescape' codec can't decode bytes in position 1-2: malformed \N character escape (<ipython-input-139-d41a2e6995ab>, line 12)

In [143]:
# make videoGame table
data = pd.read_csv('title_ketan.tsv',sep='\t')
data = data[data.titleType.isin(['videoGame'])]

file = open('videoGame.sql','w') 
file.close()
file = open('videoGame.sql','a') 
# generate title table
for index, row in data.iterrows():
    string = "Insert into videoGame values( "
    for col in VIDEOGAME_COLUMN_NAME:
        if row[col] == '\\N' or row[col] == None:
            string += 'NULL' +", "
        if type(row[col]) != str:
            string += "" + str(row[col]) +", "
        else:
            string += "'" + row[col] + "', "
    
    string = string[:-2]
    string += " ); \n"
    file.writelines(string)
file.close() 
data.to_csv('videoGame_ketan.tsv', sep='\t')

In [127]:
title.titleType.unique()

array(['tvEpisode', 'tvMovie', 'movie', 'tvSeries', 'short', 'video',
       'videoGame', 'tvSpecial', 'tvShort', 'tvMiniSeries'], dtype=object)

In [144]:
# make tvEpisode table
data = pd.read_csv('title_ketan.tsv',sep='\t')
data = data[data.titleType.isin(['tvEpisode'])]

file = open('episodes.sql','w') 
file.close()
file = open('episodes.sql','a') 
# generate title table
for index, row in data.iterrows():
    string = "Insert into episodes values( "
    for col in VIDEOGAME_COLUMN_NAME:
        if row[col] == '\\N' or row[col] == None:
            string += 'NULL' +", "
        if type(row[col]) != str:
            string += "" + str(row[col]) +", "
        else:
            string += "'" + row[col] + "', "
    
    string = string[:-2]
    string += " ); \n"
    file.writelines(string)
file.close() 
data.to_csv('tvEpisode_ketan.tsv', sep='\t')

In [74]:


title = title[['tconst','titleType', 'primaryTitle', 'isAdult', 'startYear', 'genres']].tail(10000)
title.dropna(inplace=True)
title = pd.DataFrame(title.genres.str.split(',').tolist()).stack()
title = title.reset_index()[['tconst','titleType', 'primaryTitle', 'isAdult', 'startYear']]
title.columns = [['tconst','titleType', 'primaryTitle', 'isAdult', 'startYear', 'genres']]


KeyError: "['titleType' 'primaryTitle' 'isAdult' 'startYear' 'genres'] not in index"